File: test-suite-kotlin-ksp/src/test/kotlin/io/micronaut/docs/http/server/executeon/PersonController.kt
package io.micronaut.docs.http.server.executeon

// tag::imports[]
import io.micronaut.docs.http.server.reactive.PersonService
import io.micronaut.docs.ioc.beans.Person
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.scheduling.TaskExecutors
import io.micronaut.scheduling.annotation.ExecuteOn
// end::imports[]

// tag::class[]
@Controller("/executeOn/people")
class PersonController (private val personService: PersonService) {

    @Get("/{name}")
    @ExecuteOn(TaskExecutors.IO) // <1>
    fun byName(name: String): Person {
        return personService.findByName(name)
    }
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/docs/http/server/executeon/PersonController.py
# tag::imports[]
from micronaut.docs.http.server.reactive.PersonService import PersonService
from micronaut.docs.ioc.beans.Person import Person
from micronaut.http.annotation import Controller, Get
from micronaut.scheduling import TaskExecutors
from micronaut.scheduling.annotation import ExecuteOn
# end::imports[]


# tag::class[]
@Controller("/executeOn/people")
class PersonController:
    def __init__(self, personService: PersonService):
        self.personService = personService

    @Get("/{name}")
    @ExecuteOn(TaskExecutors.IO)  # <1>
    def byName(self, name: str) -> Person:
        return self.personService.findByName(name)
# end::class[]
